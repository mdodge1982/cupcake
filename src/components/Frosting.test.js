import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import Frosting from './Frosting';

test('Initial render', () => {
	const frostingArr = [{id:123}];
	const component = renderer.create(
		<Frosting frosting={frostingArr} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Triggers angleShoot on click', () => {
	const mockFn = jest.fn();
	const frostingArr = [{id:123}]
	const component = ReactTestUtils.renderIntoDocument(
		<Frosting angleShoot={mockFn} frosting={[]} />
	);
	const ul = ReactTestUtils.findRenderedDOMComponentWithTag(component,'ul');
	ReactTestUtils.Simulate.click(ul);
	expect(mockFn.mock.calls.length).toBe(1);
});
