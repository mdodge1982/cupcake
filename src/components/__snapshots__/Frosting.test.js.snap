// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial render 1`] = `
<ul
  className="Frosting"
  onClick={[Function]}
>
  <li
    className="FrostingBlob"
    style={
      Object {
        "bottom": "undefinedpx",
        "height": "undefinedpx",
        "left": "undefinedpx",
        "transform": "rotate(NaNrad)",
        "width": "undefinedpx",
      }
    }
  />
</ul>
`;
